addons/account/models/account_invoice.py

```diff
@@ -192,25 +192,27 @@ def _get_payments_vals(self):
             return []
         payment_vals = []
         currency_id = self.currency_id
+        if self.type in ('out_invoice', 'in_refund'):
+            matched_field = 'matched_debit_ids'
+            partial_field = 'debit_move_id'
+        elif self.type in ('in_invoice', 'out_refund'):
+            matched_field = 'matched_credit_ids'
+            partial_field = 'credit_move_id'
```

Comment:
Why not something a bit more readable like:
```
if self.type in ('out_invoice', 'in_refund'):
    partials = self.payment_move_line_ids.mapped('matched_debit_ids').filtered(lambda partial: partial.debit_move_id.move_id == self)
else:
    partials = self.payment_move_line_ids.mapped('matched_credit_ids').filtered(lambda partial: partial.credit_move_id.move_id == self)

amount = sum(partials.mapped('amount')
amount_currency = sum(partials.mapped('amount_currency'))
payment_currency_ids = set(partial.currency_id for partial in partials)
payment_currency_id = next(payment_currency_ids) if len(payment_currency_ids) == 1 or False
```

Reply:
![lKPaHzp](https://user-images.githubusercontent.com/17759223/65976648-bc695300-e470-11e9-8ddd-5b5b51f956c9.jpg)

well, because we do the loop each time we map or sum
Sometimes the loop could be huge